src/inverse-kinematics/src/ConvexHullHelpers.cpp

```diff
@@ -342,5 +342,20 @@ namespace iDynTree
         return margin;
     }
 
+    Vector2 ConvexHullProjectionConstraint::projectAlongGravity(iDynTree::Position& posIn3dInAbsoluteFrame)
+    {
+        iDynTree::Vector2 projected;
+        toEigen(projected) = toEigen(Pimu)*toEigen(posIn3dInAbsoluteFrame-o);
+        return projected;
+    }
+
+    void ConvexHullProjectionConstraint::setPimu(const iDynTree::Direction xProjection, const iDynTree::Direction yProjection, const iDynTree::Position originOfPlaneInWorld)
```

Comment:
This method is meant to set the rows of the projection matrix, that are not constained to be orthonormal. However, the `iDynTree::Direction` class is meant to represent direction in the 3D space, and so its content are forced to be normalized (see https://github.com/robotology/idyntree/blob/2470a71cee49ee3a611195e36b8dc447eca9f154/src/core/src/Direction.cpp#L23). 
This will create deformation in the projection logic. 
To avoid this problem, I suggest to modify the API to remove the `Pimu` attribute and the `setPimu` method and just have a single method `ConvexHullProjectionConstraint::projectAlongDirection` that takes the direction along which to project the com everytime, and will compute the Pimu on the fly.